### ISSUE TITLE
8BitDo Ultimate C 2.4g Wireless Controller not working

### ISSUE THREAD

**Issue by Scdk** [o0Zz/sys-con#21]
First of all thanks continuing the project!!

**Information**
Atmosphere version: 1.6.2
Switch OS version: 17.0.1
Syscon version: 1.2.1
Controller: 8BitDo Ultimate C 2.4g Wireless Controller

**Description**
I connect the controller in both d-input or x-input mode and even though the light of the controller stays on, indicating a successful connection, the controller is not recognized by the switch. Neither buttons, analog sticks of dpad works and the controller doesn't even show up in the controllers menu.

I tried to verify the HID code for the 8bitdo in the device manager, but when I was going to insert it into the config file I noticed that it was already there (2dc8-3106). I noticed that in x-input mode the button layout resemble more the xbox template than the xbox360, but even changing this configuration the controller still not works.

When I enabled the log level 0 I got the message "No HID or XBOX interfaces found !" in the logs.

Both my config and log files are in the zip attached to the issue.

**Logs**
[logs.zip](https://github.com/user-attachments/files/15948891/logs.zip)

**Comment by o0Zz** [comment]
HI, yeah the profile don't have any impact regarding the detection, it will have an impact only for the configuration. Here the switch don't detect controller at all. What happen if you do exactly the same thing on a PC ? Does your PC detect the controller ? I assume you tried it with the dongle, could you try just to plug it with a USB cable to see if it change something ? Thanks

**Comment by o0Zz** [comment]
Any progress on this issue ?

**Comment by Scdk** [comment]
The logs previously sent were in fact made using the dongle. I had made tests with the cable but unfortunately the controller didn't even show signs of connection, the center LED kept blinking, indicating it was searching for connection.

My PC with Windows detects the controller normally, both with the dongle and the cable.

The log file of my tests can be found bellow.

[log.log](https://github.com/user-attachments/files/15980302/log.log)

**Comment by texita** [comment]
May i ask if you have Mission control enabled?

**Comment by o0Zz** [comment]
Ok sounds to be a controller issue,. I did some research on the net and people have same issue than you with various systems (linux etc ...) a workaround is to force the system to detect this controller with vid/pid and associate the correct driver. 

So let's try it, edit you configuration file like that:
`
[global]
discovery_mode=1
discovery_vidpid=2dc8-3106
`

Reference:
https://gist.github.com/ammuench/0dcf14faf4e3b000020992612a2711e2

Let us know if its better.

**Comment by Scdk** [comment]
I had mission control enabled, just to be sure I disabled it. 

Unfortunately even with the edited configuration file the controller still wasn't detected.

The log file of my tests can be found bellow.

[log.log](https://github.com/user-attachments/files/15997240/log.log)

**Comment by texita** [comment]
> I had mission control enabled, just to be sure I disabled it.
> 
> Unfortunately even with the edited configuration file the controller still wasn't detected.
> 
> The log file of my tests can be found bellow.
> 
> [log.log](https://github.com/user-attachments/files/15997240/log.log)


Just to be sure, have you tried with mission control disabled and discovery_mode=0? (untouched config.ini)

**Comment by Scdk** [comment]
I disabled the mission control via hekate toolbox and restarted the switch, I even tried to connect my ps4 controller but it didn't work so motion control was in fact not working.

Just to be sure I attached both my config and log files bellow. But I tried both discovery mode 0 and 1 with discovery_vidpid=2dc8-3106.

I tried both cable and dongle in both modes, the cable doesn't seem to send any signal to the switch, it just charges the controller.

I updated the firmware of both the dongle and the controller and tried every combination again, but unfortunately had the same result. The combinations wore: cable, X input dongle, D input dongle and all those again with discovery mode in 0.

In the 8BitDo Ultimate C 2.4g Wireless Controller the change between X mode and D mode are done by turning the controller on by pressing the X button or the B button. I tried on my PC and could see that the controller was in fact changing modes.


[config_and_log.zip](https://github.com/user-attachments/files/16022580/config_and_log.zip)

**Comment by texita** [comment]
Thanks for your tests, i explain why asked you this for everyone:

Because of the behavior of no logs when a pad is connected was similar to ps3 pads when mission control is enabled, i personally tested that. The problem is the bluetooth, have a look on this issue, is still valid for this sys-con https://github.com/ndeadly/MissionControl/issues/782
So i was thinking your pad got bluetooth feature because of this issue https://github.com/ndeadly/MissionControl/issues/651 if you read the last comment from ndeadly he's not sure C version of this pad is without bluetooh.
So best thing to do is try your pad without Mission control enabled until we get it working.

At this point i don't know why you get empty logs, more questions that i can ask:
have you got other controllers working? like ps4 pad is working with sys-con? 
other installed modules?
without sys-con is your controller working on the switch?

Maybe with these informations @o0Zz can think something

**Comment by o0Zz** [comment]
Hi,
4 questions here:
1) Could you try this verion and let us know if it change something: 
[sys-con-1.2.2-ATMOSPHERE-1.6.x-custom.zip](https://github.com/user-attachments/files/16025023/sys-con-1.2.2-ATMOSPHERE-1.6.x-custom.zip)
2) Could you let us know if you installed any driver on your windows PC ? 
3) Could you send a screenshot of your 'joy.cpl' and your 'Device Manager' to try to understand how the controller is detected
4) Additionnaly if you can check the VID/PID reported by windows

Thanks

**Comment by Scdk** [comment]
Hi, I'm happy to help with the tests, I hope the controller works with sys-con but even if it doesn't I want to at least help figure out why this happens.

I believe this version of the controller really does not have an Bluetooth adapter. The naming convention of the 8bitdo controllers is very confusing, but as one of the cheapest versions i believe this one truly only supports 2.4g connection.


Answering to texita questions:

- The ps4 controller works flawlessly with sys-con, mission control is still disabled, and just to confirm I disabled sys-con and the controller stooped working.

- I got my switch already hacked so even I don't know all the modules installed. I annexed pictures of my hbmenu and hakate toolbox just to be sure that nothing that is installed could be interfering with sys-cons's functionalities.

- Without sys-con enabled the controller ironically behaves just like in my previous tests. With the connection via dongle in both modes and cable not recognizing any inputs. Just like before the dongle shows signs of connection but the cable seems to just charge the controller.


Now answering to o0Zz questions:

1 - I've installed the 1.2.2 version of the module on it's release day, unfortunately it didn't solve the controller's problem. In fact I believe most of my tests wore made in this version.

2 - I didn't install any drivers in my pc, both the connection by dongle and cable worked from the get go.

3 - I annexed bellow screenshots of both my joy.cpl and device manager in all modes of connection.

4 - Seeing the screenshots of the previous question I realized that each mode of connection has a different VID/PID, those being [2dc8-3015] for cable, [2dc8-3016] for dongle in d-input mode and (the one that was used in the previous tests) [2dc8-3106] for dongle in x-input mode. Just to be sure I added the new VID/PID to my config mode and tested all modes of connection again. The results unfortunately wore the same. I also tested setting the discovery mode to 1 and doing all the tests with both the d-input and cable VID/PID. Unfortunately none of those strategies worked. 

The log of my tests and the final config file can also be found bellow.

[config-logs-images.zip](https://github.com/user-attachments/files/16060631/config-logs-images.zip)

**Comment by texita** [comment]
Thanks for all the informations. First, i was reading and linked the wrong FAQ above, for the non C version, so sorry, anyway the situation not changing.
from the FAQ i read this:


have you tried this without sys-con enabled? (be careful what they say about settings section on step 1) At least to get it working with it's original function :) If you get it working without sys-con enabled, then let's go back with sys-con enabled.
Plus i wanna ask if you tried the custom version of 1.2.2 that o0Zz attach in his latest post, not the original 1.2.2 on  the release section

**Comment by texita** [comment]
Thanks for all the informations. First, i was reading and linked the wrong FAQ above, for the non C version, sorry for that, this is the right one https://support.8bitdo.com/faq/ultimate-c-2.4g-wireless-controller.html so this controller will never work without sys-con enabled. Anyway the situation is not changing for now.
Second i don't see any module that should interfere with sys-con, except for Mission Control (IF your controller has got bluetooth of course, but at this point i agree with you and i don't think so)

from the FAQ i read this, have you tried on the switch? (with sys-con enabled)
The controller is not pairing with its 2.4G receiver, what should I do?
Press and hold [X+home] or [B+home] to turn on the controller, plug the 2.4G receiver into your device, hold the minus button on the controller for 5 seconds to re-pair the controller and its receiver, the home LED starts to blink rapidly, then wait for the reconnection between them.

i think you have to use only X+HOME, as in the FAQs they say B+HOME is for android
Is the controller plug-and-play?
Yes. Please plug the dongle into the USB port of your Windows device, then press the home button to play. When it’s on Android or Raspberry Pi device, plug the dongle into the USB port of the device, then press and hold [B+home] to turn it on.

i've seen the images you attach, in windows if you test the controller properties and press buttons (where you see red circles) is it working in every mode? I mean, if you press buttons you see red circles lighting up? I'm thinking it should only work only in x-input mode: [2dc8-3106] dongle x-input mode and maybe with [2dc8-3015] for cable. With [2dc8-3016] dongle in d-input it should work only in android/raspberry right?

Plus i wanna ask if you tried the custom version of 1.2.2 that o0Zz attach in his latest post, not the original 1.2.2 on the release section.

**Comment by o0Zz** [comment]
@Scdk I think you didn't tried to version I attached to this ticket, this one exactly -> https://github.com/user-attachments/files/16025023/sys-con-1.2.2-ATMOSPHERE-1.6.x-custom.zip

This binary is diffrent from the v1.2.2 released in this github. It's a custom binary built just for your with some additionnal code in order to try to better understand your issue. From the log you sent, it seems you are still running the official release and not this One exactly. Could you please confirm and try this one exactly (By clicking on the link here). thanks

**Comment by Scdk** [comment]
Hi, sorry I didn't realize it was an altered version.

Unfortunately even after the installation of the new version the controller still refuses to respond to any input in any of the modes. I've even tried with discovery mode 1 but the controller still behaved the same way. As always I will attach the config and logs bellow.

As for holding the minus button to reconnect the controller to the dongle, I hadn't tested, but when trying the controller didn't even blink it's lights or showed other signs of a search for connection. I think that this only works if the controller has no connection to the dongle.

When connected to the windows all modes works in the "joy.cpl". In d-input the button layout changes, but some circle always turn red when a button is pressed.

My roommate also has a controller exactly like mine, just to be sure I also realized the tests in his controller and got the same results.

One thing that I thought could influence sys-con is the Pro Controller Wired Communication configuration. I recently bought an 8BitDo Ultimate 2.4G Controller and in its FAQ it is stated that for it to work in the switch this option must be turned on. Mine was always on through all the tests but for today I turned it off in some tests just to see if I made any difference, unfortunately it didn't.

[config-log.zip](https://github.com/user-attachments/files/16075046/config-log.zip)

**Comment by o0Zz** [comment]
Ok That's crazy, the switch don't reconize at all this USB device. I'm not sure to understand why, we will do a last test:

```
[global]
discovery_mode=1
discovery_vidpid=2dc8-*
```

We updated the discovery_vidpid. From your screenshot you provided, we can see that depending how you plug your controller (Via dongle, via usb, via d-input  or x-input) there is multiple PID. Thus all tests we did by the past was based on "dongle usb x-input" with PID=3106 if i'm correct.
Let's try to ask the switch to discover all VID=2dc8 and see when you plug it with USB Cable or d-input if it works.

Thanks for doing all these tests ;)

**Comment by Scdk** [comment]
It was my pleasure helping with the tests. Unfortunately the last ones still didn't produce any response from the controller.

Yesterday I had tried each of the VID/PID found in the screenshots, but to no avail.

As always, my config and log files can be found bellow.

[config-log.zip](https://github.com/user-attachments/files/16091296/config-log.zip)

**Comment by o0Zz** [comment]
Ok nothing really great here. Do you have another controller with you other than the 8bit one ? If yes, could you try to connect it and see if it works?

Indeed I wondering if this issue is not caused by another problem with your switch (exemple: OTG adaptor not working, other system module installed etc ...) Because i'm not able to figure out why the controller is not detected at all. Indeed when we look at the log, the switch don't notify at all something has been plugged, like if you never plugged anything.

**Comment by texita** [comment]
@o0Zz He already tried that, working. 

> Answering to texita questions:
> 
> * The ps4 controller works flawlessly with sys-con, mission control is still disabled, and just to confirm I disabled sys-con and the controller stooped working.

What about a wireshark capture ath this point? https://github.com/o0Zz/sys-con/blob/master/doc/WiresharkCapture.md

**Comment by o0Zz** [comment]
Ok, I didn't see that answer. Wireshark capture won't help because it's at the low level of the Nintendo Switch that the controller is not detected, meaning that Horizon OS doesn't provide the information to sys-con to be able to address this controller. So either we have to play with the detection functions (which we tried a bit with pid/vid) or maybe wait for a Nintendo Switch update (FW 18 ?)

**Comment by o0Zz** [comment]
After a while I changed my mind :), If you can provide a wireshark capture. I will try to update the switch detection mode in order to detect your controller by "InterfaceProtocol" instead of InterfaceClass or bcdDevice, we will see if it change something

**Comment by gvilardefarias** [comment]
Hi, I have this same controller and I'm having this same issue.

**Comment by o0Zz** [comment]
Ok. could you do make a  [wireshark capture](https://github.com/o0Zz/sys-con/blob/master/doc/WiresharkCapture.md). and share it in order to see if there is something to do.

**Comment by gvilardefarias** [comment]
I attached one wireshark usb capture, I put the receiver then started the controller and after I pressed a few buttons and removed the receiver. 

[8bit ultimate C usb capture.zip](https://github.com/user-attachments/files/16475315/8bit.ultimate.C.usb.capture.zip)

**Comment by o0Zz** [comment]
Ok could you try this in your configuration file:

```
[global]
discovery_mode=1
discovery_vidpid=2dc8-*
```

The original post mention that the pid of the device is 3106 while it's in fact 3016 according to the wireshark capture.

**Comment by gvilardefarias** [comment]
I tried it but it didn't work even plugging in by cable, Maybe this can help: https://gist.github.com/ammuench/0dcf14faf4e3b000020992612a2711e2

**Comment by o0Zz** [comment]
Thanks for the share. Unfortunatly the problem in this post is a bit different than your. Like #30 I will try to simulate this controller on my side to understand what the switch dont like and try to find a workaround but it will not be quick I will need some time to do it. I will keep you updated.

**Comment by o0Zz** [comment]
Hi,

Quick question, Do you use the Switch Dock ? I means, when you did these tests, did you plug this controller to the dock or did you used a USB-C adapter ?
If you only tried Dock, could you try to use a USB-C adapter and see if this controller is working ?

Thanks

**Comment by gvilardefarias** [comment]
Hi, Tried it and didnt work.

**Comment by o0Zz** [comment]
Thank you for your reply. I did some investigation on my side in order to better understand the switch USB discovery process:
https://github.com/o0Zz/sys-con/blob/master/doc/ReverseEngineerUSB.md
For now I don't really understand what's wrong here.
Do you have any USB hub with an external power supply (Like this: https://www.amazon.co.uk/Powered-RSHTECH-Aluminum-4-port-Individual-Black/dp/B08ZKSK6MB)? Or do you have a way to have 1 in order to make a test ? I would like to check if the issue would be a "power consumption" issue.
In other words the idea would be to plug the powered USB hub to the switch and plug the controller on it.

**Comment by o0Zz** [comment]
Thank you for the all the information and investigations you provided in this ticket. I believe we can close this issue for now, marking it as unresolved. It appears there's some incompatibility between the Switch and this controller, but it doesn’t seem to be related to the controller’s profile. I simulated your controller on my side using the Wireshark capture you provided, and everything worked properly. This suggests the controller profile is not at fault, but rather there may be an issue at a lower level between the Switch and the  controller. The only potential solutions I see are either updating your controller (if possible - You can also try to downgrade it the idea here is to change the controller behavior to make the switch happy) or waiting for Nintendo to modify the Switch's behavior at low level which seems unlikely.

**Comment by o0Zz** [comment]
Hi guys, I re-open this ticket since @VaflyaYT found a way to discover the controller on switch ! I think we still have some work to do but at least sys-con have a way to intercept the controller.

We tried many combinaisons in this ticket but one of them is to turn off hid discovery and only keep xbox discovery without any custom vid/pid

```
[global]
discovery_mode=1
discovery_vidpid=
```

I assume the controller need to be in XInput mode (@VaflyaYT could you confirm ?).
And magically the controller will appear and will immediatly disapear because I assume sys-con is not providing the correct boot sequence (that can probably be fixed).

Before going further in deep debugging, can one of you try below combinaisons and provide the log for every one ?

```
[2dc8-3106]
profile=xboxone
```


```
[2dc8-3106]
driver=generic
```
**Important note** here we are using "driver=" and not "profile=" that's what we want here

Thanks

**Comment by VaflyaYT** [comment]
> I assume the controller need to be in XInput mode (@VaflyaYT could you confirm ?)
Yes,i used XInput

**Comment by VaflyaYT** [comment]
> Before going further in deep debugging, can one of you try below combinaisons and provide the log for every one ?

[log.log](https://github.com/user-attachments/files/18288790/log.log)

**Comment by o0Zz** [comment]
You didn't set the log as "trace": https://github.com/o0Zz/sys-con?tab=readme-ov-file#logs that make the investigation difficult. But anyway, I just realized from your previous logs that the controller is in fact a xboxone and not xbox360. 
Try this version, it change the profile of your controller from xbox360 to xboxone and add more logs (I also already enabled trace mode)
[sys-con-1.5.0+3.zip](https://github.com/user-attachments/files/18288996/sys-con-1.5.0%2B3.zip)

Another note, from the last logs you provided and the old one, I noticed that the controller behave differently, i'm not sure to understand exactly what's happen (if you do something different) but sometime we get this error:

**Error 1**
```
|E|00:00:22.207|3994E070| Controller[2dc8-3106] Failed to open output endpoint 4 !
```
and sometimes:
**Error 2**
```
|E|00:01:05.637|58D4E070| SwitchUSBEndpoint[0x05] Write failed: 0x0003228C
```

If you get error 1, try to understand what is happening, and try to re-iterate until you are able to open the endpoint, you should see:

```
|D|00:03:29.674|3994E070| SwitchUSBEndpoint[0x05] Successfully opened !
```

**Comment by VaflyaYT** [comment]
> Another note, from the last logs you provided and the old one, I noticed that the controller behave differently, i'm not sure to understand exactly what's happen (if you do something different) but sometime we get this error:

I tried connecting the controller by wire via otg and in the dock, maybe it has some effect.
[log.log](https://github.com/user-attachments/files/18289192/log.log)
I found only error 2 in the logs, but the controller is not displayed in HOS

**Comment by o0Zz** [comment]
Ok even with a xboxone init sequence this controller refuse to work.
Is it possible to get a wireshark capture of this controller in XInput Mode (https://github.com/o0Zz/sys-con/blob/master/doc/WiresharkCapture.md ) ? What I need is what's happen when you plug this controller to your Windows or Linux PC - The idea is to capture the traffic between your PC and the controller like I have a reference that works and I can simulate the same behavior on the Switch.

**Comment by VaflyaYT** [comment]
just in case, I did it through a wire and through a 2.5g key (maybe they are somehow different).
[8bitdo ultimate C 2.4g.zip](https://github.com/user-attachments/files/18289590/8bitdo.ultimate.C.2.4g.zip)

**Comment by o0Zz** [comment]
Thanks for sharing the dump !
After diving into it, I have to say—this controller is quite weird. It advertises an Xbox One profile but sends Xbox 360 data. That said, none of this fully explains why the Switch cannot successfully send write data. I’m starting to wonder if the controller might not be fast enough to process incoming data.

To investigate further, I’ve created a modified version that introduces a delay before writing and attempts multiple writes based on what I observed in your dump

[Download sys-con-1.5.0+4.zip](https://github.com/user-attachments/files/18298903/sys-con-1.5.0%2B4.zip)

Could you test it in two scenarios—one with a wired connection and another with the USB dongle? 
It would also help immensely if you could split the logs for each test to simplify debugging and comparison.

Thanks

**Comment by VaflyaYT** [comment]
[wired.log](https://github.com/user-attachments/files/18302399/wired.log)
[dongle.log](https://github.com/user-attachments/files/18302400/dongle.log)
when I checked, the controller did not appear in the HOS

**Comment by o0Zz** [comment]
Thanks, we will skip the dongle for now since we can't even open it, we will only focus on wired.

So, this version force syscon to only detect xbox360 controller (and not xboxone controller) I would like to see if by forcing the switch to only open xbox360 it will change something ->

[Uploading sys-con-1.5.0+5.zip…]()

**Comment by VaflyaYT** [comment]
I can't download it.

**Comment by o0Zz** [comment]
Done

**Comment by VaflyaYT** [comment]
[log.log](https://github.com/user-attachments/files/18306086/log.log)

**Comment by o0Zz** [comment]
Ok Nothing is detected, I'm not really surprised. Can you do below tests with this new version:
[sys-con-1.5.0+6.zip](https://github.com/user-attachments/files/18306344/sys-con-1.5.0%2B6.zip)

Test 1: Use XInput mode, plug the controller then get the logs
Test 2: Use DInput mode, plug the controller then get the logs

--- Here update the /config/sys-con/config.ini and change discovery_mode=0 by discovery_mode=1

Test 3: Use XInput mode, plug the controller then get the logs
Test 4: Use DInput mode, plug the controller then get the logs

**Comment by VaflyaYT** [comment]
[discovery_mode=1.log](https://github.com/user-attachments/files/18306419/discovery_mode.1.log)
[discovery_mode=0.log](https://github.com/user-attachments/files/18306420/discovery_mode.0.log)

**Comment by o0Zz** [comment]
Could you clarify when you used DInput (the specific timeframe)? From the log, I only see references to XInput, so I’m unsure when you changed for DInput.

Additionally, could you share the firmware version of your controller?

I noticed a discrepancy in behavior. @Scdk and @gvilardefarias tested the same configuration (with discovery_mode=0), but their results were very different—nothing was detected from the Switch. On your side, however, the same configuration detects something but disconnects immediately. I’m trying to understand why this difference occurs, as it might help in identifying the root cause of the issue.

**Comment by Elb1to** [comment]
Hey everyone, I know this might not come as a surprise, but the same behavior occurs with the 8BitDo Ultimate 2C Wireless Controller. The controller isn't detected using the 2.4Ghz dongle that comes with the controller, neither via USB cable...

Though, when it is connected to the USB Cable, the 8BitDo logo button stays on (like if it was successfully connected and paired).

**Comment by o0Zz** [comment]
> Hey everyone, I know this might not come as a surprise, but the same behavior occurs with the 8BitDo Ultimate 2C Wireless Controller. The controller isn't detected using the 2.4Ghz dongle that comes with the controller, neither via USB cable...
> 
> Though, when it is connected via USB cable, the 8BitDo logo button stays on (like if it was successfully connected and paired).
> 
> EDIT: Here's the Amazon link to the one I have: https://a.co/d/1K6mIYt

Hi, The Ultimate 2C is properly working according to this https://github.com/o0Zz/sys-con/issues/43.

**Comment by Elb1to** [comment]
> Hi, The Ultimate 2C is properly working according to this https://github.com/o0Zz/sys-con/issues/43.

Is there any sequence I must follow in order to get it to pair? With the dongle connected and the controller connected via USB (and set to 2.4Ghz, not Bluetooth), I do get a vibration as if it paired and the 8BitDo logo button stays on, though, I can't get it to show up as a controller on the Switch. Currently using SysCon v1.4.3

**Comment by o0Zz** [comment]
The config.ini in the v1.5.0 already include these lines, thus I'm not sure what you did but the original sys-con version should natively support this controller.

**Comment by danbarretto** [comment]
Hello, any progress on this issue? I have an Ultimate C 2.4g Wireless controller and I can help with some wireshark captures if needed.

**Comment by o0Zz** [comment]
> Could you clarify when you used DInput (the specific timeframe)? From the log, I only see references to XInput, so I’m unsure when you changed for DInput.
> 
> Additionally, could you share the firmware version of your controller?
> 
> I noticed a discrepancy in behavior. [@Scdk](https://github.com/Scdk) and [@gvilardefarias](https://github.com/gvilardefarias) tested the same configuration (with discovery_mode=0), but their results were very different—nothing was detected from the Switch. On your side, however, the same configuration detects something but disconnects immediately. I’m trying to understand why this difference occurs, as it might help in identifying the root cause of the issue.

Hi, From my last comment I wondering if someone else can test this controler with this config, to see if it works.
```
[global]
discovery_mode=1
discovery_vidpid=
```
And provide the firmware version your are using on the controller (I have the feeling the firmware version has an impact on this issue).

Additionnally, if you can provide logs when the controller is pluged with dinput and when it's plugged with xinput (in 2 separate files)
Thanks

**Comment by Scdk** [comment]
Hi, since my last comment I switched to an OLED switch that's currently running the firmware 19.0.0, atmosphere 1.8.0 and sys-con 1.5.1.

I tried the requested config and got the same results as @VaflyaYT. I first tried to connect the controller by dongle in Dinput mode, then by cable and finally, when I tried by dongle in Xinput I got a popup informing me that a new controller was connected. However, none of the buttons did anything, which probably means that it was disconnected right after, just as mentioned before.

My config and log files can be found below.

[log.zip](https://github.com/user-attachments/files/19239693/log.zip)

**Comment by o0Zz** [comment]
Your logs are different from @VaflyaYT, the controller is not discovered at all - At least from what I see.
Do you have a way to get your controller firmware version ?
Did you try the XInput mode by cable ?

**Comment by andrewvarial** [comment]
**Information**
Atmosphere version: 1.11.2
Switch OS version: 22.5.0
Syscon version: 1.6.3
Controller: 8BitDo Ultimate C 2.4G Wireless Controller

**Description**
The controller doesn't work with the console. I've tried using the cable and the included dongle, but nothing worked.

I think the issue is specifically with the controller itself. I also have an 8BitDo Ultimate Controller, which is natively supported by Switch via Bluetooth, BUT by using it with 2.4 GHz XInput mode it connects as USB controller with its dongle with no problem.

The firmware on both controllers is up to date, and I have no issues with them on PC.

**Log**
[log.txt](https://github.com/user-attachments/files/29818556/log.txt)

**Comment by o0Zz** [comment]
Yes, this controller isn’t recognized by the Switch. My assumption is that 8BitDo implemented something in the firmware that prevents the Switch from detecting it. I’m not sure whether this was intentional or simply an issue on their side, but the Ultimate C 2.4G is one of the few controllers that doesn’t work with sys-con.
